### ISSUE TITLE
[member-delimiter-style] Enforce None Option

### ISSUE THREAD

**Issue by Skillz4Killz** [typescript-eslint/typescript-eslint#672]
**Repro**

```js
{
  "rules": {
    '@typescript-eslint/member-delimiter-style': {
      "multiline": {
        "delimiter": "none",
        "requireLast": false
      },
      "singleline": {
        "delimiter": "comma",
        "requireLast": false
      }
    }
  }
}
```

```TS
export interface RawUser {
	id: string;
	username: string,
	discriminator: string,
	avatar: string | null,
	bot?: boolean,
	mfa_enabled?: boolean;
	locale?: string,
	verified?: boolean
	email?: string
}
```

**Expected Result**
I expected the linter remove the semis and commas. For example, if you set semis and save the file everything gets semis added to it
**Actual Result**
No errors. I have semis, commas and none mixed in. Still nothing is fixed.

**Additional Info**
The purpose of this rule is to enforce a style. There are cases where it is best to enforce no semis or commas for example for projects using the Standard Eslint which does not enforce semis. Having semis in typings with this rule makes that really annoying. 

The word to disable this rule should be `off` whereby this rule is simply disabled similar to all other rules. The word `none` should be used to ENFORCE the no semis nor commas. Or maybe someone can think of a better name as opposed to none for requiring there be no semis and no commas

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `"^1.11.0"` |
| `@typescript-eslint/parser`        | `"^1.11.0"` |
| `TypeScript`                       | `"^3.4.5"` |
| `ESLint`                           | `"^6.0.1"` |
| `node`                             | `v11.14.0` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
please read the eslint documentation on configuring rules
https://eslint.org/docs/user-guide/configuring#configuring-rules

**Comment by Skillz4Killz** [comment]
@bradzacher I apologize if I am confused but I don’t believe the issue has anything to do with configuring rules. The `semi` option enforces it properly as per the documentation. However, when choosing `none` option it does not enforce it anymore. 

![image](https://user-images.githubusercontent.com/23035000/60744278-360da580-9f43-11e9-9505-c332c6fe4e7e.png)
![image](https://user-images.githubusercontent.com/23035000/60744291-4887df00-9f43-11e9-9157-58c20491f32c.png)

Based on this documentation, it should be enforcing that there be line breaks to separate. Switching to semi does work. Switching to none disabled the rule instead of enforcing the line breaks.

**Comment by bradzacher** [comment]
I was responding in regards to the config you posted.

```diff
 {
   "rules": {
-    '@typescript-eslint/member-delimiter-style': {
+    '@typescript-eslint/member-delimiter-style': ['error', {
       "multiline": {
         "delimiter": "none",
         "requireLast": false
       },
       "singleline": {
         "delimiter": "comma",
         "requireLast": false
       }
-     }
+     }]
   }
 }
```

If this isn't the exact config you are using, then we can reopen this, and look into it.

---

Though there are tests that are passing in master which suggest that the rule is working as expected..?

https://github.com/typescript-eslint/typescript-eslint/blob/1d9ca845d5af4bb4eeee8b41719a182ae131fc56/packages/eslint-plugin/tests/rules/member-delimiter-style.test.ts#L1055-L1250

https://github.com/typescript-eslint/typescript-eslint/blob/1d9ca845d5af4bb4eeee8b41719a182ae131fc56/packages/eslint-plugin/tests/rules/member-delimiter-style.test.ts#L1605-L1880

https://github.com/typescript-eslint/typescript-eslint/blob/1d9ca845d5af4bb4eeee8b41719a182ae131fc56/packages/eslint-plugin/tests/rules/member-delimiter-style.test.ts#L2162-L2357

https://github.com/typescript-eslint/typescript-eslint/blob/1d9ca845d5af4bb4eeee8b41719a182ae131fc56/packages/eslint-plugin/tests/rules/member-delimiter-style.test.ts#L2712-L2987

**Comment by Skillz4Killz** [comment]
I have tried with and without the `error` This is how the entire eslintric file looks.

```js
module.exports = {
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint'],
  extends: 'bamboo',
  rules: {
    // TODO: Remove this once complete so we can stop using bad ways typescript does not recommend using namespaces
    '@typescript-eslint/no-namespace': 'off',
    // Disables requiring semi colons
    '@typescript-eslint/semi': 'off',
    // This rule changes it to no longer require semi colons on typings
    '@typescript-eslint/member-delimiter-style': ['error', {
      "multiline": {
        "delimiter": "none",
        "requireLast": false
      },
      "singleline": {
        "delimiter": "comma",
        "requireLast": false
      }
    }],
    // This rule makes it so that by default functions are accessible publicly
    '@typescript-eslint/explicit-member-accessibility': 'no-public'
  }
};
```

Unfortunately, with or without the `[‘error’` and reloading VSC window, I see no errors. 

![image](https://user-images.githubusercontent.com/23035000/60745587-124d5e00-9f49-11e9-9d25-a0336cced76a.png)
